projects/extension/src/background/ConnectionManager.ts

```diff
@@ -246,7 +290,9 @@ export class ConnectionManager
     if (!this.#client) {
       throw new Error("Smoldot client does not exist.")
     }
-    const { name, id, relay_chain } = JSON.parse(chainSpec)
+    const { name, id, relay_chain }: SpecInterface = JSON.parse(
+      chainSpec,
+    ) as SpecInterface
```

Comment:
This `SpecInterface` thing seems wrong. You have no guarantee that a chain spec passed by the user actually has these fields. Which is also a good reason why shouldn't use the `name` anywhere. I don't know how TypeScript behaves if a field is missing, but if it throws an exception then it's definitely wrong.

Reply:
This is wrong - did not see it - thank you for pointing out.
The SpecInterface should be altered to having optional params (e.g. name and relay_chain) but id should always be there